erpnext/compliance/taxes.py

```diff
@@ -23,6 +23,8 @@ def calculate_cannabis_tax(doc):
 		# calculate cultivation tax for buying cycle
 		cultivation_taxes = calculate_cultivation_tax(doc)
 		for account, tax in cultivation_taxes.items():
+			if not tax:
+				continue
```

Comment:
This won't work, I mentioned this in the other PR too. If you suddenly go from taxed to non-taxed (in any way), your old tax row won't get deleted. The deletion needs to happen post calculating the taxes and setting the rows.

Reply:
@vishdha please fix this